metrics/csv_metrics.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 The Celo Authors
+// This file is part of the celo library.
+//
+// The celo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The celo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the celo library. If not, see <http://www.gnu.org/licenses/>.
+
+package metrics
+
+import (
+	"encoding/csv"
+	"io"
+	"os"
+	"sync"
+)
+
+// A CSVRecorder enables easy writing of CSV data a specified writer.
+// The header is written on creation. Writing is thread safe.
+type CSVRecorder struct {
+	writer        csv.Writer
+	backingWriter io.Writer
+	writeMu       sync.Mutex
+}
+
+// NewStdoutCSVRecorder creates a CSV recorder to standard out
+// The header is immediately written upon construction.
+func NewStdoutCSVRecorder(fields ...string) *CSVRecorder {
```

Comment:
I would remove this function since we're not using and there isn't another clear reason to keep it.

Reply:
done